### ISSUE TITLE
Is DbFlow compatible with Gradle 3 ?

### ISSUE THREAD

**Issue by Tboy70** [agrosner/DBFlow#1572]
Hi, 

I would like to know if DbFlow is compatible with Gradle 3.0 ? 

Indeed, I tried to migrate my gradle to 3.1.2. So, I must use implementation and annotationProcessor instead of compile and atp.

However, when I run the app, FlowManager and FlowConfig are not found. 

Is it normal or am I doing something wrong ? 

DBFlow Version: 4.2.4

Thank you

**Comment by hannesa2** [comment]
I use 4.2.4 with Gradle build tools 3.1.2, for use of differnt flavors with different package names you need 
```
        <provider
            android:name="com.raizlabs.android.dbflow.runtime.StubContentProvider"
            android:authorities="${applicationId}.authority"
            android:exported="false"/>
```
use it during init

<img width="984" alt="image" src="https://user-images.githubusercontent.com/3314607/39400697-f0b779a0-4b35-11e8-8c54-c6a9ccafc65e.png">


and use it here

<img width="848" alt="image" src="https://user-images.githubusercontent.com/3314607/39400691-d54a6f9c-4b35-11e8-9ae1-fc26d1c38d26.png">

hope this will help you

**Comment by hannesa2** [comment]
I can just guess what's wrong.
For me it looks like you have DBFlow in library and you use it in :app ?
I'm not surprised, when you change 'implementation' of DBFlow with outdated and deprecated 'compile'. --> But this is a bad solution

I'm not sure what's the best solution:
You can add dbflow in :app as well ? --> I would avoid this

I would solve this in a BaseApplication located in library with DBFlow init and Application :app extends from it. --> But what's about you use dbflow in more places in :app ?

merge library and :app ?

Let us know about your solution

**Comment by carlosbrolo** [comment]
Same issue as Tboy70 ....   Android studio is not allowing anymore the use of compile.
My partial sollution is to write:
`annotationProcessor "com.github.Raizlabs.DBFlow:dbflow-processor:${dbflow_version}"
api "com.github.Raizlabs.DBFlow:dbflow-core:${dbflow_version}"
api "com.github.Raizlabs.DBFlow:dbflow:${dbflow_version}"`

Then all the code I had works, even when marked with error (compilation/build). The downside is now I see errors everywhere and delays my codding.

**Comment by agrosner** [comment]
fixed in 5.0.0-alpha1
